test/image/mocks/bar_text_relative.json

```diff
@@ -0,0 +1,40 @@
+{
+    "data":[
+        {
+            "width":[1,0.8,0.6,0.4],
+            "text":[1,2,3,4],
+	    "textposition":"outside",
+            "y":[1,2,3,4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":[0.4,0.6,0.8,1],
+            "text":["Three",2,"This bar is size one",0],
+	    "textposition":"auto",
+            "y":[3,2,1,0],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "width":1,
+            "text":[-1,-3,-2,-4],
+	    "textposition":"inside",
+            "y":[-1,-3,-2,-4],
+            "x":[1,2,3,4],
+            "type":"bar"
+        }, {
+            "text":[0,"very small size",-3,-2],
```

Comment:
> Shall I rotate it like in the case of inside labels?

Yes, I think so. One concern with this (it's already a concern but will get worse when we rotate the text) is whether we need to include that label in autorange (via `Axes.expand` and its `ppadplus` and `ppadminus` options). That could get awfully annoying though, as you'd have to make the outside text elements available for sizing during `setPositions` rather than `plot`, but then you still need to know the bar width to tell whether to rotate or not... ugh. @etpinard thoughts?

> 1px padding, shall I increase it (the padding in the ggplot2 example looks too much, though)?

Agreed about ggplot2. For hover text [we use 3px](https://github.com/plotly/plotly.js/blob/master/src/plots/cartesian/constants.js#L64) - how about that?